sklearn/tests/test_calibration.py

```diff
@@ -376,6 +377,31 @@ def decision_function(self, X):
     calibrated_clf.fit(X, y)
 
 
+def test_isotonic_calibration_strict():
+    """
+    Test that enforcing strict monotonicity for isotonic calibration
+    preserves rank-based metrics to handle issue
+    https://github.com/scikit-learn/scikit-learn/issues/16321
+    """
+
+    n_samples = 100
+    X, y = make_classification(n_samples=2 * n_samples, n_features=6,
+                               random_state=42)
+    X_train, y_train = X[:n_samples], y[:n_samples]
+    X_test, y_test = X[n_samples:], y[n_samples:]
+    clf = LogisticRegression(C=1.)
+    clf.fit(X_train, y_train)
+    y_pred = clf.predict_proba(X_test)
+    roc_pred = roc_auc_score(y_test, y_pred[:, 1])
+
+    calibrated_clf_strict = CalibratedClassifierCV(clf, method='isotonic',
+                                                   cv='prefit', strict=True)
+    calibrated_clf_strict.fit(X_train, y_train)
+    y_pred_calib = calibrated_clf_strict.predict_proba(X_test)
+    roc_calib_strict = roc_auc_score(y_test, y_pred_calib[:, 1])
+    assert_almost_equal(roc_pred, roc_calib_strict)
```

Comment:
testing that the ranking metrics are unchanged is a good check, but it's mostly a side effect.

We should also explicitly check `IsotonicREgression` that strict monotonicity was enforced. I think this can be done by checking the results of `argmax` with a stable sort (e.g. mergesort)?

Reply:
Good point ! I'll add a specific test for that.